src/core/packages/pricing/common/src/types.ts

```diff
@@ -36,4 +36,22 @@ export interface IPricingTiersClient {
    * @returns True if the feature is available in the current pricing tier, false otherwise
    */
   isFeatureAvailable<TFeatureId extends string>(featureId: TFeatureId): boolean;
+
+  /**
+   * @deprecated Don't rely on this API for customizing serverless tiers. Register a dedicated feature and use `isFeatureAvailable` instead.
+   */
+  product(): PricingProduct | undefined;
+}
+
+interface PricingProductSecurity {
+  type: 'security';
+  tier: 'search_ai_lake' | 'complete' | 'essentials';
+  product_lines: Array<'ai_soc' | 'endpoint' | 'cloud'>;
+}
+
+interface PricingProductObservability {
```

Comment:
Don't want to block the PR on this, it just bothers me that these product-specific definitions live at core level.
It would be awesome if we could define this at solution level somehow.